File: langs/java/src/test/java/com/github/tetrisanalyzer/piecegenerator/DefaultPieceGeneratorTest.java
package com.github.tetrisanalyzer.piecegenerator;

import com.github.tetrisanalyzer.settings.TetrisAnalyzerGameSettings;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class DefaultPieceGeneratorTest {

    @Test
    public void nextPiece() {
        PieceGenerator pieceGenerator = new DefaultPieceGenerator(5, new TetrisAnalyzerGameSettings());

        String result = "";

        for (int i=0; i<20; i++) {
            result += pieceGenerator.nextPiece();
        }

        assertEquals("TLOIJJSIJJIOZJZZTLZT", result);
    }
}

File: langs/scala/src/test/scala/nu/tengstrand/tetrisanalyzer/piecegenerator/DefaultPieceGeneratorTest.scala
package nu.tengstrand.tetrisanalyzer.piecegenerator

import nu.tengstrand.tetrisanalyzer.BaseTest
import org.junit.Test

class DefaultPieceGeneratorTest extends BaseTest {

  @Test def nextPiece(): Unit = {
    val pieceGenerator = new DefaultPieceGenerator(5)
    val pieces = (1 to 20).foldLeft("") { (result,n) => result + pieceGenerator.nextPiece().toString }

    pieces should be ("OLIZTTSZTTZIJTJJOLJO")
  }
}
